fix(Settings): write settings to disk when they change

vith opened the pull request:
This prevents losing settings changes when qTox is terminated abnormally,
such as by power loss or crash.

issue #1969

I have another branch in progress with unix signal handlers to quit gracefully when the system is shutting down as well.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/3192)

<!-- Reviewable:end -->

sudden6 commented on src/persistence/settingstransaction.cpp:
```diff
@@ -0,0 +1,114 @@
+ */
+SettingsTransaction::SettingsTransaction()
+{
+//    qDebug() << "created" << this;
```
Commented out Debug statements shouldn't be in the final code. Just remove them with another commit so they are visible in the git log.

vith committed:
fix(Settings): write settings to disk when they change

This prevents losing settings changes when qTox is terminated abnormally,
such as by power loss or crash.

issue #1969

vith committed:
fix(Settings): remove debug noise

vith commented:
Done.

sudden6 commented:
This looks ok at the first glance, but will probably take some time to review thoroughly

vith commented:
Had some discussion on IRC. Todo:
- [ ] Add a throttle/debounce timer so SettingsTransaction can be used on things like window resize events.

zetok commented:
#2733 ?

zetok commented:
Hi,

As you might be aware, repo has been moved under `qTox` org on github.

What does it have to do with your PR, you might ask.

Well, it doesn't change all that much, except that workflow has been slightly improved – now there are required checks that PR needs to pass in order to be merged.

Those checks are:
- travis (checks whether PR builds & commit message format)
- reviewable (requirement for PR to be reviewed & accepted)

Since osx travis was broken for a bit, you will need to rebase your PR on newest master to get it to pass travis check. Sorry for inconvenience.

Reminder: as always, PR needs to be possible to merge without conflicts. Make sure it does ;)

iphydf commented:
@vith are you going to rebase this branch?

zetok commented:
PR didn't have any activity for over 2 months, and no one seems to be interested in working on it, thus closing.

If that's not the case, please either ask to reopen the pull, or make a new one.
